single-line-diagram-force-layout/src/main/java/com/powsybl/sld/force/layout/ForceLayout.java

```diff
@@ -0,0 +1,122 @@
+package com.powsybl.sld.force.layout;
+
+import org.jgrapht.Graph;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+
+public class ForceLayout {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ForceLayout.class);
+
+    private static final double DEFAULT_REPULSION = 400.0;
+    private static final double DEFAULT_DAMPING = 0.5;
+    private static final double DEFAULT_MAX_SPEED = Double.POSITIVE_INFINITY;
+    private static final double  MIN_ENERGY_THRESHOLD = 0.00001;
+
+    private final double repulsion;
+    private final double damping;
+    private final double maxSpeed;
+
+    public ForceLayout() {
+        repulsion = DEFAULT_REPULSION;
+        damping = DEFAULT_DAMPING;
+        maxSpeed = DEFAULT_MAX_SPEED;
+    }
+
+    // TODO: make graph’s node and edge generics instead of point and spring type?
+    public void execute(Graph<Point, Spring> graph, int maxSteps) {
+        double deltaTime = 0.01;
+        int iterationCounter = 0;
+        boolean isStopped = false;
+
+        while (!isStopped) {
+            this.applyCoulombsLaw(graph);
+            this.applyHookesLaw(graph);
+            this.attractToCenter(graph);
+            this.updateVelocity(graph, deltaTime);
+            this.updatePosition(graph, deltaTime);
+
+            iterationCounter++;
+            if (this.totalEnergy(graph) <= MIN_ENERGY_THRESHOLD) {
```

Comment:
Can we avoid unnecessary iterations by putting the threshold a bit higher? Do you think it would be interesting to also put that threshold in the constructor? so that the user can parameterize the convergence criteria.